mlir-sycl/lib/Dialect/SYCL/IR/SYCLOps.cpp

```diff
@@ -610,6 +610,28 @@ LogicalResult SYCLHostScheduleKernel::verify() {
   return verifyNdRange(*this, range, offset, ndRange);
 }
 
+void SYCLHostScheduleKernel::getEffects(
+    SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
+        &effects) {
+  auto *defaultResource = SideEffects::DefaultResource::get();
+  // The nd-range arguments will be read from
+  if (Value range = getRange())
+    effects.emplace_back(MemoryEffects::Read::get(), range, defaultResource);
+  if (Value offset = getOffset())
+    effects.emplace_back(MemoryEffects::Read::get(), offset, defaultResource);
+  // The rest of the arguments will be scalar or have rw access
+  for (Value arg : getArgs()) {
+    if (!isa<LLVM::LLVMPointerType>(arg.getType())) {
+      // TODO: When we have accessor type information, drop write access for
```

Comment:
TBH, I'm still not quite happy with this comment. For accessors, IMHO we want to drop the write access for *all* accessors, independent of whether they are marked read-only or not. 

The only reason we can't do this right now is that the operation might be optimized away if we do so. So the first part of the comment should be updated to reflect this.

Reply:
I added memory effects not attached to any argument so that we model the operation writes to global memory.